code/examples/channel/src/state.rs

```diff
@@ -114,19 +121,31 @@ impl State {
             }
 
             let hash = hasher.finalize().to_vec();
-            let signature =
-                signature.expect("Fin part is guaranteed to exist due to checks on lines 90-92");
+            let Some(signature) = signature else {
+                error!(proposer = %parts.proposer, "Fin proposal part is missing");
+                return Ok(None);
+            };
+
+            // Get proposal's public key
+            let public_key = self
+                .get_validator_set()
+                .validators
+                .iter()
+                .find(|v| v.address == parts.proposer)
```

Comment:
I just realized we can shorten this by doing:

```suggestion
                .get_by_address(&parts.proposer)
```